File: python/java_to_python/09_pgm_avion_interface/src/co/edu/campusucc/poo/app/MainAirships.py
from src.co.edu.campusucc.poo.implement import TipoAvioneta, TipoCarga, TipoComercial, TipoMilitar
from src.co.edu.campusucc.poo.interfaces import IAvionStandar

if __name__ == "__main__":
    
    avion = TipoComercial()
    avion.setMarca("Boeing")
    avion.setTipo("Comercial")
    print(avion)

    avion.startingEngine()
    avion.speedUp()
    avion.descent()
    avion.currentSpeed()
    avion.landing()
    avion.stopingEngine()

    avion = TipoCarga()
    avion.setMarca("Airbus")
    avion.setTipo("Carga")
    print(avion)

    avion.startingEngine()
    avion.speedUp()
    avion.speedUp()
    avion.currentSpeed()
    avion.descent()
    avion.descent()
    avion.landing()
    avion.stopingEngine()

    avion = TipoMilitar()
    avion.setMarca("Lockheed Martin")
    avion.setTipo("Aereonave de Combate")
    print(avion)

    avion.startingEngine()
    avion.speedUp()
    avion.speedUp()
    avion.currentSpeed()
    avion.shootMissiles()
    avion.descent()
    avion.descent()
    avion.landing()
    avion.stopingEngine()

    avion = TipoAvioneta()
    avion.setMarca("Tecnam P2002 Sierra Mk2")
    avion.setTipo("Avion de Agricultura")
    print(avion)

    avion.startingEngine()
    avion.speedUp()
    avion.speedUp()
    avion.currentSpeed()
    avion.irrigation()
    avion.descent()
    avion.descent()
    avion.landing()
    avion.stopingEngine()

File: ruby/pgms_ruby/09-pgmAvionInterface/MainAirships.rb
require 'TipoAvioneta'
require 'TipoCarga'
require 'TipoComercial'
require 'TipoMilitar'
require 'IAvionStandar'

if __FILE__ == $0
  avion = TipoComercial.new
  avion.setMarca("Boeing")
  avion.setTipo("Comercial")
  puts avion

  avion.startingEngine
  avion.speedUp
  avion.descent
  avion.currentSpeed
  avion.landing
  avion.stopingEngine

  avion = TipoCarga.new
  avion.setMarca("Airbus")
  avion.setTipo("Carga")
  puts avion

  avion.startingEngine
  avion.speedUp
  avion.speedUp
  avion.currentSpeed
  avion.descent
  avion.descent
  avion.landing
  avion.stopingEngine

  avion = TipoMilitar.new
  avion.setMarca("Lockheed Martin")
  avion.setTipo("Aereonave de Combate")
  puts avion

  avion.startingEngine
  avion.speedUp
  avion.speedUp
  avion.currentSpeed
  avion.shootMissiles
  avion.descent
  avion.descent
  avion.landing
  avion.stopingEngine

  avion = TipoAvioneta.new
  avion.setMarca("Tecnam P2002 Sierra Mk2")
  avion.setTipo("Avion de Agricultura")
  puts avion

  avion.startingEngine
  avion.speedUp
  avion.speedUp
  avion.currentSpeed
  avion.irrigation
  avion.descent
  avion.descent
  avion.landing
  avion.stopingEngine
end